index.js

```diff
@@ -40,14 +64,25 @@ export default {
   subscribe(key, callback, options = {}) {
     let index = 1;
 
+    if (typeof callback !== 'function') {
+      console.error(`Registering in duix for '${key}': Callback is not a function: `, callback);
+    }
+
+    // Set the default options
+    options = {
+      onlyOnChange: false,
```

Comment:
Don’t you think that onlyOnChange should be true by default?
Because, the subscriber should actually be waiting for “a change” in order to do something.

Reply:
Which could be a scenario where a developer wants to be alerted (in the subscriber) for a `duix.set` that doesn't change the value? There may be a scenario, but, I think that scenario may be something more like "I want to know every time the `x` object is set", instead of "I want to know every time the `x` object changes".
I believe that a state manager should "manage" the `changes` on the state, so, we should probably aim to ALWAYS (I mean, with no `onlyOnChange` option at all) call the subscriber ONLY when the state changes.

One of the pending issues that are created is [Implement events](https://github.com/BrodaNoel/duix/issues/7). Where we call a subscriber every time an event happens.
If we implement it, that may be the alternative for the use case where a developer wants to be alerted (`duix.listen`) every time "something" (an event) happens (`duix.event`).

What do you think?

Reply:
Yes, I agree, I was not confortable to make that breaking change, but if you agree, I will do it (and that will simplify back the code :-) I will remove the option (and consider it as always true) and clean up this a bit... I will do that tonight probably (european time).

About the events, I don't know... let's discuss that there as for me that is a different approach... that would not impact this at first, do it?

Reply:
Delivered

Reply:
Events can wait, agree.